File: backend/travel/travel/src/main/java/com/capstone/travel/model/Travel.java
package com.capstone.travel.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Travel {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long userId;

    private String mode; // e.g., "Car" or "Bike"
    private double distanceTravelled; // in kilometers
    private double mileage; // in km per liter
    private String fuelType; // "Diesel" or "Petrol"
    private LocalDate date;


}

File: FrontEnd/src/app/models/Travel.ts


export class Travel {
    userId: number;                
    mode: string;                  
    distanceTravelled: number;     
    mileage: number;               
    fuelType: string;            
    date: string;     

    constructor(
        userId: number,
        mode: string,
        distanceTravelled: number,
        mileage: number,
        fuelType: string,
        date: string
    ) {
        this.userId = userId;
        this.mode = mode;
        this.distanceTravelled = distanceTravelled;
        this.mileage = mileage;
        this.fuelType = fuelType;
        this.date = date;
    }
}